server/src/main/java/org/elasticsearch/search/aggregations/bucket/BestBucketsDeferringCollector.java

```diff
@@ -160,14 +164,15 @@ public void prepareSelectedBuckets(long... selectedBuckets) throws IOException {
             Query query = isGlobal ? new MatchAllDocsQuery() : searchContext.query();
             weight = searchContext.searcher().createWeight(searchContext.searcher().rewrite(query), ScoreMode.COMPLETE, 1f);
         }
+
         for (Entry entry : entries) {
             final LeafBucketCollector leafCollector = collector.getLeafCollector(entry.context);
-            DocIdSetIterator docIt = null;
-            if (needsScores && entry.docDeltas.size() > 0) {
+            DocIdSetIterator scoreIt = null;
+            if (needsScores) {
```

Comment:
I'm assuming that since we're lazily creating these structures, we should never have an `entry` with empty `docDeltasBuilder`?  Should we put an assert here to make sure?

Reply:
I added a check in the Entry ctr: https://github.com/elastic/elasticsearch/pull/43339/commits/ecea58ad2be2b55317c16a8ebcd29ba18af11e20